sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala

```diff
@@ -343,6 +343,54 @@ class ExplainSuite extends ExplainSuiteHelper with DisableAdaptiveExecutionSuite
       assert(getNormalizedExplain(df1, FormattedMode) === getNormalizedExplain(df2, FormattedMode))
     }
   }
+
+  test("Explain formatted output for scan operator for datasource V2") {
```

Comment:
can we add a `table-scan-explain.sql` to test it? It's easier to see the result.

Reply:
@cloud-fan Agree. Actually i had tried but could not get the V2 scan set up through SQL. Could you please tell me how to do it ?

Reply:
Oh I see. Currently DS v2 scan is enabled only in `DataFrameReader`, so we can't get it through pure SQL. Then this is fine.

Reply:
+1. Thank you.

Reply:
Yea, I think so...